lib/features/manage_account/presentation/profiles/identities/identities_controller.dart

```diff
@@ -219,16 +221,44 @@ class IdentitiesController extends BaseController {
     );
   }
 
-  void _deleteIdentityAction(Identity identity) {
+  Future<void> _deleteIdentityAction(Identity identity) async {
     popBack();
 
     final session = accountDashBoardController.sessionCurrent;
     final accountId = accountDashBoardController.accountId.value;
     if (accountId != null && session != null && identity.id != null) {
+      await _dereferencePublicAssets(session, accountId, identity);
       consumeState(_deleteIdentityInteractor.execute(session, accountId, identity.id!));
     }
   }
 
+  Future<void> _dereferencePublicAssets(
+    Session session,
+    AccountId accountId,
+    Identity identity) async {
+    final publicAssetIds = identity
+      .signatureAsString
+      .publicAssetIdsFromHtmlContent;
+    final removeIdentityFromPublicAssetsInteractor 
+      = getBinding<RemoveIdentityFromPublicAssetsInteractor>();
+    if (removeIdentityFromPublicAssetsInteractor == null
+      || publicAssetIds.isEmpty) 
+    {
+      return;
+    } else {
+      try {
+        await removeIdentityFromPublicAssetsInteractor.execute(
+          session,
+          accountId,
+          identityId: identity.id!,
+          publicAssetIds: publicAssetIds
+        ).last;
+      } catch (e) {
+        logError('$runtimeType::_dereferencePublicAssets(): exception: $e');
+      }
```

Comment:
why not use consumeState for it

Reply:
We need to make sure this call finish before delete Identity, else it will fail of invalid identity (race condition)

Reply:
So let's use consumeState to handle the next actions when we finish consumingState. Then we won't need try/catch like that and it will be easier to handle errors.

Reply:
Done